src/main/resources/graphql/service/workPath.graphqls

```diff
@@ -0,0 +1,13 @@
+type WorkPath {
+    id: UUID!
+    name: String!
+    chapterId: UUID!
+    stages: [Stage!]!
+}
+
+type Stage{
+    id: UUID!
+    position: Int!
+    requiredContents: [Content!]!
```

Comment:
I would've liked a more generic solution instead of just storing a list of required and a list of optional contents, but @PaulBredl would probably hate me for it

Reply:
Don't we also need a way to store what counts as "completed"? E.g. setting a minimum correctness score for a quiz. Or are we storing that in the quiz service?

Reply:
You should be able to derive completeness from the content. 
Note, current implementation only contains necessary data to create and modify Work Paths & Stages.
We will add User specific data later in following sprints

Reply:
As far as I can remember our consensus was that the definition of completeness should be settable in the stage and not the content itself (as content may be re-used in mulitple stages).

But even if not, a differentiation betwee requiredContent and optionalContent does not make sense in any case because if a content is optional you could just set its minimum correctness to 0.

@PaulBredl opinions?

Reply:
> I would've liked a more generic solution instead of just storing a list of required and a list of optional contents, but @PaulBredl would probably hate me for it

@myluki2000 What do mean with more generic solution? Isn't this implementation what we discussed on Monday?

Reply:
> Don't we also need a way to store what counts as "completed"? E.g. setting a minimum correctness score for a quiz. Or are we storing that in the quiz service?

Stored in quiz service

Reply:
> But even if not, a differentiation betwee requiredContent and optionalContent does not make sense in any case because if a content is optional you could just set its minimum correctness to 0.

That doesn't make sense to me? How does a media content like a pdf have a correctness?

Reply:
> That doesn't make sense to me? How does a media content like a pdf have a correctness?

It doesn't but it has some way of deciding what counts as done. In case of a quiz it's correctness, in case of a PDF it's if it has been read or not

Reply:
> @myluki2000 What do mean with more generic solution? Isn't this implementation what we discussed on Monday?

Didn't we say on monday that the lecturer should be able to set in the stage what counts as completed or not?

In any case, having BOTH a minimum correctness in the quiz service and an "optional content" here in the content service is a duplication to me, they both map to the same concept and should not both exist at the same time. I.e. why should an optional quiz have a "minimum passing grade" when you don't need to pass it in the first place?

Reply:
> In any case, having BOTH a minimum correctness in the quiz service and an "optional content" here in the content service is a duplication to me, they both map to the same concept and should not both exist at the same time. I.e. why should an optional quiz have a "minimum passing grade" when you don't need to pass it in the first place?

No that is not a duplication at all. It actually makes more complex things possible, like saying you need 50% in this quiz and 70% in the other to complete the course.
Also you cannot make a quiz optional by making the requiredCorrectness 0. That doesn't make any sense? Even in an optional quiz you want to have feedback if you passed the quiz or not. If you just set it to 0 means:
1) in order the complete it you would still have to click through it once (akward)
2) even if I click everything wrong, the feedback would still be "Yay you completed the quiz, you're amazing" and the learning interval will be higher, etc.

Reply:
> Didn't we say on monday that the lecturer should be able to set in the stage what counts as completed or not so the student can progress to the next stage?

Yes by marking which contents are required and which are optional